javascript/app-event-analysis/src/components/EffectResult.tsx

```diff
@@ -0,0 +1,125 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { Stack, Text } from '@fluentui/react'
+import { memo, useRef } from 'react'
+import { ErrorBoundary } from 'react-error-boundary'
+import { useRecoilValue } from 'recoil'
+
+import {
+	ChartOptionsState,
+	CheckedUnitsState,
+	EventNameState,
+	OutcomeNameState,
+	PlaceboSimulationState,
+	TreatedUnitsState,
+} from '../state/state.js'
+import type { OutputData } from '../types.js'
+import {
+	BAR_CHART_HEIGHT_PERC_OF_WIN_HEIGHT,
+	BarChartOrientation,
+} from '../types.js'
+import { BarChart } from './BarChart.js'
+import { ChartErrorFallback } from './ChartErrorFallback.js'
+import { useDynamicChartDimensions } from './DimensionedLineChart.hooks.js'
+import { DimensionedLineChart } from './DimensionedLineChart.js'
+import type { EffectResultProps } from './EffectResult.types.js'
+import { useSynthControlBarChartData } from './EffectResultPane.hooks.js'
+import { TreatedTitle } from './ResultPane.styles.js'
+import Spacer from './style/Spacer.js'
+import { Container, Strong } from './style/Styles.js'
+
+export const EffectResult: React.FC<EffectResultProps> = memo(
+	function EffectResult({
+		ref,
+		treatedUnit,
+		inputData,
+		outputData,
+		hoverInfo,
+		synthControlData,
+		checkableUnits,
+		onRemoveCheckedUnit,
+	}) {
+		const eventName = useRecoilValue(EventNameState)
+		const chartOptions = useRecoilValue(ChartOptionsState)
+		const outcomeName = useRecoilValue(OutcomeNameState)
+		const treatedUnits = useRecoilValue(TreatedUnitsState)
+		const isPlaceboSimulation = useRecoilValue(PlaceboSimulationState)
+		const checkedUnits = useRecoilValue(CheckedUnitsState)
+
+		const { showChartPerUnit } = chartOptions
+
+		const barChartRef = useRef<HTMLDivElement | null>(null)
+		const barChartDimensions = useDynamicChartDimensions(
+			barChartRef,
+			BAR_CHART_HEIGHT_PERC_OF_WIN_HEIGHT,
+			synthControlData,
+		)
+
+		const synthControlBarChartData =
+			useSynthControlBarChartData(synthControlData)
+
+		const output =
+			outputData.find(o => o.treatedUnit === treatedUnit) || ({} as OutputData)
+
+		const filteredOutput = {
+			...output,
+			output_lines_control: output.output_lines_control.filter(l =>
```

Comment:
will l possibly be empty?

Reply:
I don't think so, because, in theory, the output always contains the `treatedUnit`  we're filtering.